apps/studio/components/interfaces/Realtime/Policies/RealtimePoliciesChannelRow.tsx

```diff
@@ -0,0 +1,123 @@
+  onSelectPolicyDelete: (s: PostgresPolicy) => void
+}
+
+export const PolicyRow = ({ policy, onSelectPolicyEdit, onSelectPolicyDelete }: PolicyRowProps) => {
+  const { name, command } = policy
+  return (
+    <div className="group">
+      <Panel.Content className="flex justify-between gap-2 border-b border-overlay py-4">
+        <div className="flex flex-col gap-3 lg:flex-row lg:items-center">
+          <div className="font-mono text-xs text-foreground-lighter">{command}</div>
+          <div className="flex flex-col gap-2 lg:flex-row">
+            <span className="truncate text-sm text-foreground">{name}</span>
+          </div>
+        </div>
+        <DropdownMenu>
+          <DropdownMenuTrigger asChild>
+            <Button
+              type="default"
+              style={{ paddingLeft: 4, paddingRight: 4 }}
+              icon={<MoreVertical />}
+            />
+          </DropdownMenuTrigger>
+          <DropdownMenuContent side="bottom" align="end">
+            <DropdownMenuItem className="space-x-2" onClick={() => onSelectPolicyEdit(policy)}>
+              <IconEdit size={14} />
+              <p>Edit</p>
+            </DropdownMenuItem>
+            <DropdownMenuSeparator />
+            <DropdownMenuItem className="space-x-2" onClick={() => onSelectPolicyDelete(policy)}>
+              <IconTrash size={14} />
+              <p>Delete</p>
+            </DropdownMenuItem>
+          </DropdownMenuContent>
+        </DropdownMenu>
+      </Panel.Content>
+    </div>
+  )
+}
+
+type RealtimePoliciesChannelRowProps = {
+  channel: RealtimeChannel
+  policies: PostgresPolicy[]
+  onSelectPolicyAdd: (t: string) => void
+  onSelectPolicyEdit: (p: PostgresPolicy) => void
+  onSelectPolicyDelete: (s: PostgresPolicy) => void
+}
+
+export const RealtimePoliciesChannelRow = ({
+  channel,
+  policies,
+  onSelectPolicyAdd,
+  onSelectPolicyEdit,
+  onSelectPolicyDelete,
+}: RealtimePoliciesChannelRowProps) => {
+  return (
+    <Panel
+      title={
+        <div className="flex w-full items-center justify-between">
+          <div className="flex items-center flex-row space-x-4">
+            <Antenna className="text-foreground-light w-4 h-4" />
+            <h4 className="text-lg">{channel.name}</h4>
+          </div>
+          <div className="flex space-x-1">
+            <Button type="outline" onClick={() => onSelectPolicyAdd('channels')}>
+              New policy
+            </Button>
+            <Button type="outline" onClick={() => onSelectPolicyAdd('broadcasts')}>
+              New broadcast policy
+            </Button>
+            <Button type="outline" onClick={() => onSelectPolicyAdd('presences')}>
+              New presence policy
+            </Button>
+          </div>
+        </div>
+      }
+    >
+      {policies.length === 0 ? (
+        <div className="p-4 px-6">
+          <p className="text-sm text-foreground-lighter">No policies created yet</p>
+        </div>
```

Comment:
lots of opportunity to make components for empty states, headers, layouts and son on